src/plugins/data/public/ui/filter_bar/filter_editor/value_input_type.tsx

```diff
@@ -147,6 +147,17 @@ class ValueInputTypeUI extends Component<Props> {
     this.props.onChange(boolValue);
   };
 
+  private onNumberChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const params = event.target.value;
+    this.props.onChange(
+      typeof params === 'string'
+        ? parseFloat(params)
+        : typeof params === 'bigint'
+        ? (params as BigInt).toString()
+        : params
```

Comment:
nit: instead of using a nested ternary extract part of it out, use an if/else statement, or use parens in the nested ternary to make it easier to read.

Reply:
Changed to if else statement